llama_index/indices/knowledge_graph/retrievers.py

```diff
@@ -199,8 +239,10 @@ def _retrieve(
         # add relationships as Node
         # TODO: make initial text customizable
         rel_initial_text = (
-            "The following are knowledge triplets "
-            "in the form of (subset, predicate, object):"
+            f"The following are knowledge triplets in max depth"
+            f" {self.graph_store_query_depth} "
+            f"in the form of "
+            f"`subject [predicate, object, predicate_next_hop, object_next_hop ...]`"
```

Comment:
Sure, it could be, we need to revisit this and at least introduce more mechanisms to refine, deduplicate and truncate, with your reminding, I will add in this round of push and will create e2e demo apps with it after it is merged, then I/we will come with more ideas/experiences to polish here, too.

Reply:
Makes sense!